lib/oauth2-client.js

```diff
@@ -118,8 +117,11 @@ async function _requestAccessToken({url, body, httpsAgent}) {
       body,
       agent: httpsAgent
     }));
-  } catch(e) {
-    logger.error('Error getting access token.', {error: e});
+  } catch(error) {
+    logger.error('Error getting access token.', {error});
+    if(isUnrecoverableError({error})) {
+      bedrock.exit();
```

Comment:
Some logging along the lines `...an unrecoverable error was encountered, exiting` would be good immediately before calling `bedrock.exit()`.

Reply:
Added.